pkg/prometheus/promcfg.go

```diff
@@ -1053,10 +1052,11 @@ func (cg *ConfigGenerator) generateProbeConfig(
 
 		// Add configured relabelings.
 		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(m.TypeMeta, m.ObjectMeta, m.Spec.Targets.Ingress.RelabelConfigs))...)
-		relabelings = generateAddressShardingRelabelingRulesForProbes(relabelings, shards)
+	}
 
+	if m.Spec.Targets.StaticConfig != nil || m.Spec.Targets.Ingress != nil {
```

Comment:
I don't think that the if statement is required
```suggestion
```

Reply:
This is a good remark. The operator already takes care of filtering out probes without static/ingress targets [here](https://github.com/prometheus-operator/prometheus-operator/blob/4e37c62781ce19971726f5a34b073327a3860d1a/pkg/prometheus/resource_selector.go#L497-L500). But even if the check doesn't work, it shouldn't be a problem to include the relabeling configs.

Reply:
Sounds good. This is the update ffb2a35